File: packages/protocol/swift/Sources/YiruProtocol/Status/StatusClient.swift
import Foundation
import SwiftProtobuf

public struct StatusClient: Sendable {
  private let transport: any RuntimeTransport

  public init(transport: any RuntimeTransport) {
    self.transport = transport
  }

  public func get(options: RuntimeCallOptions = RuntimeCallOptions()) async throws
    -> Yiru_Runtime_V1_GetStatusResponse
  {
    let request = Yiru_Runtime_V1_GetStatusRequest()
    let response = try await transport.unary(
      RuntimeUnaryCall(
        procedure: YiruRuntimeV1StatusServiceMethods.getStatus,
        payload: try request.serializedData(),
        options: options
      )
    )
    return try Yiru_Runtime_V1_GetStatusResponse(serializedBytes: response)
  }
}

File: packages/protocol/typescript/src/status-client.ts
import { create, fromBinary, toBinary } from '@bufbuild/protobuf'

import {
  GetStatusRequestSchema,
  GetStatusResponseSchema,
  StatusService,
  type GetStatusResponse
} from '../generated/yiru/runtime/v1/status_pb.js'
import type { RuntimeCallOptions, RuntimeTransport } from './transport.js'

const GET_STATUS_PROCEDURE = `/${StatusService.typeName}/${StatusService.method.getStatus.name}`

export class StatusClient {
  private readonly transport: RuntimeTransport

  constructor(transport: RuntimeTransport) {
    this.transport = transport
  }

  async get(options?: RuntimeCallOptions): Promise<GetStatusResponse> {
    const payload = toBinary(GetStatusRequestSchema, create(GetStatusRequestSchema))
    const response = await this.transport.unary({
      method: GET_STATUS_PROCEDURE,
      payload,
      ...(options ? { options } : {})
    })
    return fromBinary(GetStatusResponseSchema, response)
  }
}
